nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/SdkPackageUpdater.cs

```diff
@@ -254,62 +254,133 @@ private static UpdateResult TryUpdateDependencyVersion(ImmutableArray<ProjectBui
 
         foreach (var buildFile in buildFiles)
         {
-            var updateAttributes = new List<XmlAttributeSyntax>();
-            var packageNodes = FindPackageNode(buildFile, dependencyName);
+            var updateNodes = new List<XmlNodeSyntax>();
+            var packageNodes = FindPackageNodes(buildFile, dependencyName);
 
             var previousPackageVersion = previousDependencyVersion;
 
             foreach (var packageNode in packageNodes)
             {
                 var versionAttribute = packageNode.GetAttribute("Version", StringComparison.OrdinalIgnoreCase)
                     ?? packageNode.GetAttribute("VersionOverride", StringComparison.OrdinalIgnoreCase);
-                if (versionAttribute is null)
+                var versionElement = packageNode.Elements.FirstOrDefault(e => e.Name.Equals("Version", StringComparison.OrdinalIgnoreCase))
```

Comment:
my guess is that Elments cannot be null, right?

Reply:
And all elements have a non-null Name property, right?

Reply:
The area is not null-annotated but my instinct was that they would just return an empty enumerable in that case, and looking at the code for XmlElementSyntax seems to support that hypothesis.